tests/unit/client.py

```diff
@@ -0,0 +1,442 @@
+password = "hil_pass1234" or os.environ.get('HAAS_PASSWORD')
+
+sess = db_auth(username, password)
+C = Client(ep, sess)  # Initializing client library
+MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
+OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
+OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
+
+
+# Following tests check if the client library is initialized correctly
+
+def test_db_auth():
+    sess = db_auth(username, password)
+    assert sess.auth == (username, password)
+
+
+class Test_ClientBase:
+    """ When the username, password is not defined
+    It should raise a LookupError
+    """
+
+    def test_init_error(self):
+        try:
+            x = ClientBase()
+        except LookupError:
+            assert True
+
+# FIX ME: The test may vary based on which backend session is used.
+#    def test_correct_init(self):
+#        x = ClientBase(ep, 'some_base64_string')
+#        assert x.endpoint == "http://127.0.0.1:8888"
+#        assert x.auth == "some_base64_string"
+
+    def test_object_url(self):
+        x = ClientBase(ep, 'some_base64_string')
+        y = x.object_url('abc', '123', 'xy23z')
+        assert y == 'http://127.0.0.1:8888/abc/123/xy23z'
+
+# For testing the client library we need a running HIL server, with dummy
+# objects populated. Following classes accomplish that end.
+# It shall:
+#       1. Configures haas.cfg
+#       2. Instantiates a database
+#       3. Starts a server on an arbitary port
+#       4. Populates haas with dummy objects
+#       5. tears down the setup in a clean fashion.
+
+
+# pytest.fixture(scope="module")
+
+def make_config():
+    """ This function creates haas.cfg with desired options
+    and writes to a temporary directory.
+    It returns a tuple where (tmpdir, cwd) = ('location of haas.cfg', 'pwdd')
+    """
+    tmpdir = tempfile.mkdtemp()
+    cwd = os.getcwd()
+    os.chdir(tmpdir)
+    with open('haas.cfg', 'w') as f:
+        config = '\n'.join([
+            '[general]'
+            'log_level = debug',
+
+            '[auth]',
+            'require_authentication = False',
+
+            '[headnode]',
+            'base_imgs = base-headnode, img1, img2, img3, img4',
+            '[database]',
+            'uri = sqlite:///%s/haas.db' % tmpdir,
+            '[extensions]',
+            'haas.ext.auth.null =',
+            'haas.ext.network_allocators.null =',
+            '[extensions]',
+            'haas.ext.switches.mock =',
+            'haas.ext.auth.null =',
+            'haas.ext.switches.nexus =',
+            'haas.ext.switches.dell =',
+            'haas.ext.switches.brocade =',
+            'haas.ext.obm.mock =',
+            'haas.ext.obm.ipmi =',
+            'haas.ext.network_allocators.null =',
+            '[haas.ext.network_allocators.vlan_pool]',
+            'vlans = 1001-1040',
+
+        ])
+        f.write(config)
+        return (tmpdir, cwd)
+
+
+def cleanup((tmpdir, cwd)):
```

Comment:
I do not remember now, but I guess I borrowed it from some python documentation about unit tests.
without which all my unit tests were failing.